### ISSUE TITLE
How to check if 32 bit CPU has hard float support?

### ISSUE THREAD

**Issue by onnimonni** [TalalMash/Unofficial-Speedify-Installer-For-OpenWrt#2]
I checked the requirements and was wondering if my hardware will be supported.

So as you remember from the issue I opened here: https://github.com/SmoothWAN/SmoothWAN/issues/179 I'm looking into 4G/5G enabled routers.

GL-X3000 is using `MediaTek Dual-core MT7981A` SOC with more info in: https://www.mediatek.com/products/home-networking/mt7621

The CPU is `MIPS1004Kc` with more info in: https://wikidevi.wi-cat.ru/MIPS32

> The 1004Kf version includes an IEEE754 compliant Floating Point Unit, supporting
both single and double precision datatypes.

From this I understand that the c variant doesn't support floating point 😢 

I assume that **hard float** in the README.md means hardware supported floating point calculation.

Is this thinking decision correct? How can I check if hard float is supported or not?

[I understood from stackexchange](https://raspberrypi.stackexchange.com/a/71240) that I could check this by:
```
$ dpkg --print-architecture
```

But this is hard to do before ordering the device.

**Comment by onnimonni** [comment]
And my friend just mentioned to me that this is MIPS processor and not ARM 🤦.

So support will definitely not be there 😄.

Sorry for the noice!

**Comment by TalalMash** [comment]
It should work out of the box, unlike Slate AX / Flint it seems to be running at 64 bit:
You can test the latest firmware with binary magic test of any executable to be sure:
![image](https://github.com/TalalMash/Unofficial-Speedify-Installer-For-OpenWrt/assets/96490382/32d40be3-0e86-496b-9d8e-60bef61ca944)

Your link points to a different SoC which is MIPS.

**Comment by onnimonni** [comment]
You're correct and I wonder how did I end up posting the wrong link there. Thanks for the help 🙇

**Comment by onnimonni** [comment]
And thanks for teaching me about this 💪.

If someone else finds this issue from Google later on here's what you need to run in your terminal to check the same result.

```bash
# Download the update file for GL-X3000. I copied the url using my browser "copy link" button
$ curl -O https://fw.gl-inet.com/firmware/x3000/release/openwrt-x3000-4.0-0404release1-1214-1702544890.bin

# Check the file type of downloaded file
$ file openwrt-x3000-4.0-0404release1-1214-1702544890.bin
openwrt-x3000-4.0-0404release1-1214-1702544890.bin: POSIX tar archive (GNU)

# Extract the downloaded tar file
$ tar -xvf openwrt-x3000-4.0-0404release1-1214-1702544890.bin

# Check the included files
$ tree .
.
├── openwrt-x3000-4.0-0404release1-1214-1702544890.bin
└── sysupgrade-glinet_gl-x3000
    ├── CONTROL
    ├── kernel
    └── root

2 directories, 4 files

# Check the file type of root
$ file sysupgrade-glinet_gl-x3000/root
sysupgrade-glinet_gl-x3000/root: Squashfs filesystem, little endian, version 4.0, xz compressed, 58793625 bytes, 6776 inodes, blocksize: 262144 bytes, created: Thu Dec 14 08:25:22 2023

# Realize that I can't extract squasfs because I don't have unsquashfs installed for my MacOS
$ brew install squashfs

# Extract the squashfs root into root-squashfs
$ unsquashfs -f -d root-squashfs sysupgrade-glinet_gl-x3000/root

# Check the ubus binary and realize it's ARM 64 bit
$ file root-squashfs/bin/ubus
root-squashfs/bin/ubus: ELF 64-bit LSB pie executable, ARM aarch64, version 1 (SYSV), dynamically linked, interpreter /lib/ld-musl-aarch64.so.1, no section header
```

**Comment by onnimonni** [comment]
It works perfectly with GL-X3000 🥳 

1. I modified all gateway metric values from 0 to something higher.
1. I downloaded the package from your releases.
1. Followed the install instructions and installed Speedify
1. Logged in to Speedify
1. Restarted the device

I can now bond my bad VSDL and 4G directly in the router.

I'm very pleased with the results 💪

<img width="853" alt="image" src="https://github.com/TalalMash/Unofficial-Speedify-Installer-For-OpenWrt/assets/5691777/b668edcb-2174-4849-8f82-9d43e17a500f">

**Comment by onnimonni** [comment]
Before I was able to follow your instructions I needed to do this:
<img width="1306" alt="gl-inet-installation-guide" src="https://github.com/TalalMash/Unofficial-Speedify-Installer-For-OpenWrt/assets/5691777/edebac60-ebd2-47bf-a753-1d00ce8d7aad">

**Comment by TalalMash** [comment]
That's great to hear! Thanks for sharing.
Added to the list.

Renaming and reopening this issue for others to check if their router is supported.

**Comment by martinthiele** [comment]
![CleanShot 2024-02-13 at 18 01 41@2x](https://github.com/TalalMash/Unofficial-Speedify-Installer-For-OpenWrt/assets/5013868/7da1019e-d69c-4fac-a556-a539ab858034)
![CleanShot 2024-02-13 at 18 01 56@2x](https://github.com/TalalMash/Unofficial-Speedify-Installer-For-OpenWrt/assets/5013868/2d21dcac-3cc3-4f06-be20-ad36b44bc0f9)
![CleanShot 2024-02-13 at 18 02 45@2x](https://github.com/TalalMash/Unofficial-Speedify-Installer-For-OpenWrt/assets/5013868/251aa0d0-b622-421a-a378-3ee8e2bedacc)


I can not get to the speedify UI to show up - appears ngix 

This is on a Spitz  GL.iNet GL-X3000

my firmware version 

Current Firmware
Version
4.0
Firmware Type
0406release1
Compile Time
2024-01-23 15:20:08(UTC+08:00)

**Comment by leblancexplores** [comment]
@onnimonni what setup are you using? I'm trying to use GLi.Net's official OpenWRT but so far I've only got it to connect when using Tethering. It seems like with the internal 5G modem it doesn't pick up the device.

**Comment by onnimonni** [comment]
>@onnimonni what setup are you using?

This is somewhat vague question but i'll try to answer. I haven't used this for a while so I can't confirm if it works with latest firmware. You can see from the screenshots above that I was using `v4.0`.

When testing this I was using the stock antennas on [GL-X3000](https://www.gl-inet.com/products/gl-x3000/). It was connected to a local VSDL through wifi using the [Repeater mode](https://docs.gl-inet.com/router/en/4/interface_guide/internet_repeater/) and I had also installed a separate 5G sim card into the router and plugged my iPhone 12 Pro with usb cable to use tethering.

The most important thing to notice for me was this part above:
> I modified all gateway metric values from 0 to something higher.

Watch [the included video in the project README](https://github.com/TalalMash/Unofficial-Speedify-Installer-For-OpenWrt?tab=readme-ov-file#installation) for how to do that inside the openwrt dashboard.

If you read the README it seems that [Speedify is now supporting openwrt devices directly](https://support.speedify.com/article/918-openwrt). If I were you I would probably try that instead.
